src/mailmojo-widget.php

```diff
@@ -176,56 +154,60 @@ public function update ($newInstance, $oldInstance) {
 	public function widget ($args, $instance) {
 		$incname = $tags = $desc = '';
 		extract($args);
-		$mmApi = $this->mmPlugin->getApi();
-		if ($mmApi === null || empty($instance['listid'])) {
+
+		if (empty($this->mmPlugin->username) || empty($instance['listid'])) {
 			return '';
 		}
 
 		// Include name input field
 		if ($instance['incname']) {
 			$incname = <<<HTML
-<div class="field">
-	<label for="mailmojo_name">%s:</label>
-	<input class="text" type="text" id="mailmojo_name" name="mailmojo_name">
-</div>
+<p class="field">
+	<label for="mailmojo_{$this->number}_name">%s:</label>
+	<input class="text" type="text" id="mailmojo_{$this->number}_name" name="name">
+</p>
 HTML;
 			$incname = sprintf($incname, __('Name', 'mailmojo'));
 		}
 
-		// Checkboxes for tags
+		$tags = "";
 		if (!empty($instance['tags'])) {
 			$tags = $this->getHtmlForTags($instance);
 		}
 
-		// Description
+		$desc = '';
 		if (!empty($instance['desc'])) {
 			$desc = "<p>{$instance['desc']}</p>";
 		}
 
+		$tagsadditional = '';
+		if (!empty($instance['tags'])) {
+			$tagsadditional = <<<HTML
+<input type="hidden" name="tagsadditional" value="{$instance['tags']}">
+HTML;
+		}
+
 		// The main output of the widget
 		$output = <<<HTML
 {$before_widget}
 	{$before_title}{$instance['title']}{$after_title}
 	$desc
-	<form method="post" id="mailmojo_form_{$this->number}" class="mailmojo_form">
-		<div class="field">
-			<label for="mailmojo_email">%s:</label>
-			<input class="text" type="text" id="mailmojo_email" name="mailmojo_email">
-		</div>
+	<form action="https://{$this->mmPlugin->username}.mailmojo.no/{$instance['listid']}/s"
+			method="post" id="mailmojo_form_{$this->number}" class="mailmojo_form">
```

Comment:
Suggest aligning `id` attribute with the fields, so `mailmojo_{$this->number}_form`.